sdk/python/kfp/dsl/_pipeline_param.py

```diff
@@ -249,3 +249,14 @@ def ignore_type(self):
     """ignore_type ignores the type information such that type checking would also pass"""
     self.param_type = None
     return self
+
+
+def additional_input_name_for_pipelineparam(
+    param_or_name: Union[PipelineParam, str]) -> str:
+  """Gets the name for an additional (compiler-injected) input."""
+
+  # Adding a prefix to avoid (reduce chance of) name collision between the
+  # original component inputs and the injected input.
```

Comment:
Can the DAG have both explicit and implicit inputs?
AFAIK, the explicit graph inputs only exist in the root pipeline DAG and KFP's Graph components. But they never have implicit inputs.
Are you sure the collision can happen?